diff --git a/index.d.ts b/index.d.ts
index 811908a8734db3b4fb270e95ef47cc39eb6bc05c..d8fe2dafc7d40da96c9c7da3fd447d7107de0439 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -20,4 +20,4 @@
 import * as echarts from './types/dist/echarts';
 // Export for UMD module.
 export as namespace echarts;
-export = echarts;
\ No newline at end of file
+export * from './types/dist/echarts';
\ No newline at end of file
